packages/operator/pkg/validation/validation.go

```diff
@@ -41,3 +44,19 @@ func ValidateExistsInRepository(name string, repository connection.Repository) e
 	}
 	return nil
 }
+
+var idRegex = regexp.MustCompile("^[a-z0-9-]{1,63}$")
+var ErrIDValidation = errors.New("ID is not valid")
+
+// Id restrictions:
+//  * contain at most 63 characters
+//  * contain only lowercase alphanumeric characters or ‘-’
+//  * start with an alphanumeric character
+//  * end with an alphanumeric character
+func ValidateID(id string) error {
+	if idRegex.MatchString(id) && !strings.HasPrefix(id, "-") && !strings.HasSuffix(id, "-") {
+		return nil
+	}
```

Comment:
It's not obvious, can't we declare valid regexp that will cover all naming conventions and just check that it's matching?

Reply:
I am appreciate any help with the regex expression. :)

Reply:
You can use that yesterday regex :)

Reply:
@karbyshevds Thanks for help, fixed